CHANGELOG.md

```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
```

Comment:
@Scienfitz @AdrianSosic this PR is now ready to review, as Martin and I discussed that this can already be included independent of different kinds of interpoint constraints we might introduce later.